### PR TITLE
fix: Show correct login error

### PR DESCRIPTION
**Pull request by Nimasheth**
## Notion ticket link

[Show correct login error](https://www.notion.so/uwblueprintexecs/Show-correct-login-error-23bd6ae946c8403bbd43053bb3aff9d5)

when you input incorrect login info, it throws a window alert instead of showing the appropriate error text

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/auth/Login.tsx [hunk 1]
```diff
@@ -25,22 +25,31 @@ const Login = (): React.ReactElement => {
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const [loginError, setLoginError] = useState(false);
```
Comment (carissa-tang) [inline frontend/src/components/auth/Login.tsx:30]: can we rename `loginError` -> `invalidFormError`

Comment (Nimasheth) [reply frontend/src/components/auth/Login.tsx:30]: made the change

File: frontend/src/components/auth/Login.tsx [hunk 2]
```diff
@@ -25,22 +25,31 @@ const Login = (): React.ReactElement => {
...
+  const [invalidFormError, setinvalidFormError] = useState(false);
```
Comment (carissa-tang) [inline frontend/src/components/auth/Login.tsx]:
```suggestion
  const [invalidFormError, setInvalidFormError] = useState(false);
```

File: frontend/src/components/auth/Login.tsx (hunk 1 above)
Comment (carissa-tang) [reply frontend/src/components/auth/Login.tsx:30]: oh sorry i meant the old one should be named `invalidFormError` (for the unfilled fields)
and the general error is `loginError`

**Review by carissa-tang — COMMENTED** [review]
~~did design get back to you on the error message?~~

File: frontend/src/components/auth/Login.tsx (hunk 2 above)
Comment (carissa-tang) [reply frontend/src/components/auth/Login.tsx]: ^ i don't think the change was made

Comment (Nimasheth) [reply frontend/src/components/auth/Login.tsx]: i made the loginerror that says "please fill in fields" and renamed its variables to invalidformerror and setinvalidform error and then used the loginerror and setloginerror variables for the "failed to login error" so I renamed the variables! is that what you were asking?

Comment (carissa-tang) [reply frontend/src/components/auth/Login.tsx]: ohh i meant to capitalize `setinvalidFormError` -> `setInvalidFormError`

File: frontend/src/APIClients/AuthAPIClient.ts [hunk 3]
```diff
@@ -85,8 +84,7 @@ const registerTeacher = async (
       localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(user));
     }
   } catch (e: unknown) {
-    // eslint-disable-next-line no-alert
-    window.alert("Failed to sign up");
+    throw new Error("Failed to login");
```
Comment (carissa-tang) [inline frontend/src/APIClients/AuthAPIClient.ts:87]:
```suggestion
    throw new Error("Failed to sign up");
```

Comment (carissa-tang) [inline frontend/src/APIClients/AuthAPIClient.ts:87]: you also need to update `teacher-signup/index.tsx` to handle the error

**Pull request closed without merging**